README.md

```diff
@@ -1,68 +1,873 @@
+- You have a build system in place where you manage dependencies with npm
+- You do not want to load scripts directly from third party sites
+
+To install [@okta/okta-signin-widget](https://www.npmjs.com/package/@okta/okta-signin-widget):
+
+```bash
+# Run this command in your project root folder.
+[project-root-folder]$ npm install @okta/okta-signin-widget --save
+```
+
+The widget source files and assets will be installed to `node_modules/@okta/okta-signin-widget/dist`, and will have this directory structure:
+
+```bash
+node_modules/@okta/okta-signin/dist/
+  css/
+    # Main CSS file for widget styles. Try not to override the classes in this
+    # file when creating a custom theme - the classes/elements are subject to
+    # change between releases
+    okta-sign-in.min.css
+
+    # Example theme that you can use as a template to create your own custom theme
+    okta-theme.css
+
+  # Base font and image files that are used in rendering the widget
+  font/
+  img/
+
+  js/
+    # Main JS file that exports the OktaSignIn object in UMD format
+    okta-sign-in.min.js
+
+  # Localized strings that are used to display all text and labels in the
+  # widget. Three output formats are included - json, jsonp, and properties
+  labels/
+
+  # Sass files that are used to generate the widget css. If you are already
+  # using Sass in your project, you can include these helper files to make
+  # generating your custom theme easier
+  sass/
+```
+
+After running `npm install`:
+
+1. Copy the assets to a folder that will be distributed to your publicly hosted site. The folders you'll need to copy are `css`, `font`, `img`, `js` and `labels`.
+
+2. Instead of copying `js` and including it in your page as a global, you can require the Sign-In Widget in your build if you are using [Webpack](https://webpack.github.io/), [Browserify](http://browserify.org/) or another module bundling system that understands the `node_modules` format.
+
+    ```javascript
+    // Load the Sign-In Widget module
+    var OktaSignIn = require('@okta/okta-signin-widget');
+
+    // Use OktaSignIn
+    var signIn = new OktaSignIn(/* configOptions */);
+    ```
+
+# API
+
+## new OktaSignIn(config)
+
+Creates a new instance of the Sign-In Widget with the provided config options. The widget has many config options to customize and change its behavior, which you can read about in the [Configuration](#configuration) section. The only required option to get started is `baseUrl`, the base url for your Okta domain.
+
+- `config` - Options that are used to configure the widget
+
+```javascript
+var signIn = new OktaSignIn({baseUrl: 'https://acme.okta.com'});
+```
+
+## renderEl(options, success, error)
+
+Renders the widget to the DOM, and passes control back to your app through the success and error callback functions when the user has entered a success or error state.
+
+- `options`
+  - `el` - CSS selector for the container element to attach the widget to.
```

Comment:
How about "CSS selector which identifies the container element that the widget attaches to." (ending a sentence with a preposition is okay if the object of the preposition is clear, but having the first "to" makes the sentence a bit harder to read).